### PR TITLE
planner: add sqlDigest and planDigest field when show bindings

### PR DESCRIPTION
**Pull request by fzzf678**
### What problem does this PR solve?

Print `sqlDigest` and `planDigest` when show bindings, which can be used to create binding and set binding status in the future.

Issue Number: ref #39199

Problem Summary:

### What is changed and how it works?
Add two fields to mysql.bind_info.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: planner/optimize.go [hunk 1]
```diff
@@ -547,12 +547,14 @@ func handleEvolveTasks(ctx context.Context, sctx sessionctx.Context, br *bindinf
...
-		Status:    bindinfo.PendingVerify,
-		Charset:   charset,
-		Collation: collation,
-		Source:    bindinfo.Evolve,
+		BindSQL:    bindSQL,
+		Status:     bindinfo.PendingVerify,
+		Charset:    charset,
+		Collation:  collation,
+		Source:     bindinfo.Evolve,
+		PlanDigest: digestWithoutDB.String(),
```
Comment (Reminiscent) [inline planner/optimize.go:557]: Should be `SQLDigest` here?

**Comment by Reminiscent** [comment]
Please change the `digestWithoutDB` to `sqlDigestWithoutDB.

**Review by Reminiscent — APPROVED** [review]

File: planner/optimize.go (hunk 1 above)
Comment (fzzf678) [reply planner/optimize.go:557]: > Should be `SQLDigest` here?

yes, you are right

File: bindinfo/bind_test.go [hunk 2]
```diff
@@ -17,6 +17,7 @@ package bindinfo_test
 import (
 	"context"
 	"fmt"
+	utilparser "github.com/pingcap/tidb/util/parser"
```
Comment (qw4990) [inline bindinfo/bind_test.go:20]: Please move this line down to the below part.

File: session/bootstrap.go [hunk 3]
```diff
@@ -2079,6 +2084,14 @@ func upgradeToVer103(s Session, ver int64) {
 	doReentrantDDL(s, CreateStatsTableLocked)
 }
 
+func upgradeToVer104(s Session, ver int64) {
```
Comment (qw4990) [inline session/bootstrap.go:2087]: Please update all existing bindings' `sql_digest` when upgrading.

File: session/bootstrap.go [hunk 4]
```diff
@@ -1649,7 +1654,7 @@ func upgradeToVer67(s Session, ver int64) {
 
 	mustExecute(s, "DELETE FROM mysql.bind_info where source != 'builtin'")
 	for original, bind := range bindMap {
-		mustExecute(s, fmt.Sprintf("INSERT INTO mysql.bind_info VALUES(%s, %s, '', %s, %s, %s, %s, %s, %s)",
+		mustExecute(s, fmt.Sprintf("INSERT INTO mysql.bind_info VALUES(%s, %s, '', %s, %s, %s, %s, %s, %s, '', '')",
```
Comment (qw4990) [inline session/bootstrap.go:1657]: No need to update here.

**Review by qw4990 — APPROVED** [review]

**Pull request merged**